frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -242,6 +247,7 @@ func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State,
 		}
 		if d.base != nil {
 			d.state = d.base.state
+			d.platform = d.base.platform
```

Comment:
Sorry I'm not very familiar with this part, but just making sure we're not overwriting d.platform. Shouldn't this be under `if d.platform == nil` ?

Reply:
If you are inheriting from a stage you should also inherit the platform of that stage. I don't think any other case makes sense.